displays/cavity_display/cavity_display.py

```diff
@@ -71,4 +72,14 @@ def __init__(self, parent=None, *args, **kwargs):
 
         self.groupbox = QGroupBox()
         self.groupbox.setLayout(self.groupbox_vlayout)
-        self.vlayout.addWidget(self.groupbox)
\ No newline at end of file
+        self.vlayout.addWidget(self.groupbox)
+
```

Comment:
Implementation of the fault counter button looks great overall. One thing I noticed is that this button follows the same pattern as the decoder button, would it be beneficial to maybe extract both into a helper method to reduce code duplication?